python/cudf/cudf/tests/test_concat.py

```diff
@@ -931,19 +914,7 @@ def test_concat_join_no_overlapping_columns(
         axis=axis,
     )
 
-    # TODO: Remove special handling below
-    # after following bug from pandas is fixed:
-    # https://github.com/pandas-dev/pandas/issues/43584
-    assert_eq(
-        expected,
-        actual,
-        check_index_type=False
-        if sort
-        and axis == 1
-        and isinstance(expected.index, pd.Int64Index)
-        and isinstance(actual.index, gd.RangeIndex)
-        else True,
-    )
+    assert_eq(expected, actual, check_index_type=True)
```

Comment:
I'm a little confused about when we choose to use a compatibility layer like `PANDAS_LT_140` and when we just make a change that works for the pinned version of pandas. This test would fail for pandas less than 1.4.0, in my understanding. However, some other tests were marked `xfail` conditional on the pandas version compatibility.

Reply:
Versioned the checks. 👍